### PR TITLE
chore: add tests for observer authz package

### PR DESCRIPTION
**Pull request by mevan-karu**
Related to https://github.com/openchoreo/openchoreo/issues/2997

This pull request adds comprehensive unit tests for the `authz` package in the observer component, significantly increasing test coverage for core authorization logic, helper functions, and client-server interactions. The new tests cover various edge cases, error handling, and correct construction of requests for both individual and batch authorization checks.

The most important changes include:

**New Test Coverage for Client Logic:**
- Added `internal/observer/authz/client_test.go` with extensive tests for the `Client`'s construction and its `Evaluate`, `BatchEvaluate`, and `GetSubjectProfile` methods. These tests cover scenarios like invalid configuration, server errors, unreachable endpoints, malformed responses, and correct handling of authorization headers.

**New Test Coverage for Helper Functions:**
- Added `internal/observer/authz/helpers_test.go` with tests for helper functions such as `ComponentScopeAuthz`, `LogsScopeAuthz`, and `CheckAuthorization`. The tests verify correct resource type resolution, error handling for invalid inputs, and correct request construction for authorization checks.

**Test Utilities and Mocks:**
- Introduced test helpers like `noopLogger`, `newTestClient`, `newDecisionServer`, and minimal PDP mocks to facilitate isolated and reliable testing of authorization behaviors. [[1]](diffhunk://#diff-dc263027d06cd01e51a3d3952258eac48d9df4e6a14fd2212cf1a49e0d30191dR1-R373) [[2]](diffhunk://#diff-8a45844e482a94054c75725ac1ecc886f51ec7e38fd200bdf1b26e097ab2af19R1-R358)

These changes ensure the authorization logic is robust, well-tested, and maintainable.

### CODE REVIEW

File: internal/observer/authz/client_test.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
+// Copyright 2025 The OpenChoreo Authors
```
Comment (LakshanSS) [inline internal/observer/authz/client_test.go]: 2026

File: internal/observer/authz/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
+// Copyright 2025 The OpenChoreo Authors
```
Comment (LakshanSS) [inline internal/observer/authz/helpers_test.go]: 2026

File: internal/observer/authz/client_test.go (hunk 1 above)
Comment (mevan-karu) [reply internal/observer/authz/client_test.go]: Fixed.

File: internal/observer/authz/helpers_test.go (hunk 2 above)
Comment (mevan-karu) [reply internal/observer/authz/helpers_test.go]: Fixed.

File: internal/observer/authz/client_test.go [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+		w.Header().Set("Content-Length", "100") // lie about body length
+		w.WriteHeader(http.StatusOK)
+		// Hijack and close the connection without writing the body
+		hj, ok := w.(http.Hijacker)
+		if !ok {
+			return
+		}
+		conn, _, _ := hj.Hijack()
+		conn.Close()
+	}))
```
Comment (chathuranga95) [inline internal/observer/authz/client_test.go:315]: Let's not use hijack for this use case. We can simplify with:
```suggestion
    srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Length", "100")
		w.WriteHeader(http.StatusOK)
	}))
```

Comment (mevan-karu) [reply internal/observer/authz/client_test.go:315]: Fixed.
